src/ucs/config/parser.c

```diff
@@ -1234,6 +1238,71 @@ static void ucs_config_parser_mark_env_var_used(const char *name, int *added)
     pthread_mutex_unlock(&ucs_config_parser_env_vars_hash_lock);
 }
 
+static char *ucs_config_get_value_from_config_file(const char *name)
+{
+    khiter_t iter = kh_get(ucs_config_map, &ucs_config_file_vars, name);
+
+    if (iter == kh_end(&ucs_config_file_vars)) {
+        return NULL;
+    }
+
+    return kh_val(&ucs_config_file_vars, iter);
+}
+
+
+static int ucs_config_parse_config_file_line(void *arg, const char *section,
+                                             const char *name,
+                                             const char *value)
+{
+    khiter_t iter = kh_get(ucs_config_map, &ucs_config_file_vars, name);
+    int override  = *(int*)arg;
+    int result;
+
+    if (iter != kh_end(&ucs_config_file_vars)) {
+        if (override) {
+            ucs_free(kh_val(&ucs_config_file_vars, iter));
+        } else {
+            ucs_error("found duplicate '%s' in config map", name);
+            return 0;
+        }
+    } else {
+        iter = kh_put(ucs_config_map, &ucs_config_file_vars,
+                      ucs_strdup(name, "config_var_name"), &result);
+        if (result == UCS_KH_PUT_FAILED) {
+            ucs_error("inserting '%s' to config map failed", name);
+            return 0;
+        }
+    }
+
+    kh_val(&ucs_config_file_vars, iter) = ucs_strdup(value, "config_value");
+    return 1;
+}
+
+ucs_status_t ucs_config_parse_config_file(const char *path, int override)
+{
+    static uint8_t file_parsed = 0;
```

Comment:
i think better move `static uint8_t config_file_parsed` to ucs_config_parser_fill_opts function, under UCS_INIT_ONCE scope. since the "parsed" is for a specific file path, which we pass from ucs_config_parser_fill_opts ()